engine-extensions/src/main/java/io/nosqlbench/engine/extensions/computefunctions/ComputeFunctions.java

```diff
@@ -66,14 +69,16 @@ public static double recall(long[] relevant, long[] actual) {
 
     public static double recall(long[] relevant, long[] actual, int k) {
         if (actual.length < k) {
-            throw new RuntimeException("indices fewer than limit, invalid precision computation: index count=" + actual.length + ", limit=" + k);
+            logger.warn("Returned indices fewer than limit in recall calculation: index count=" + actual.length + ", limit=" + k);
         }
-        relevant = Arrays.copyOfRange(relevant,0,k);
-        actual = Arrays.copyOfRange(actual, 0, k);
+        long divisor = Math.min(relevant.length, k);
+        int arrayLength = Math.max(relevant.length, actual.length);
```

Comment:
Per our discussions on Slack, if we get a mismatch between array length of KNN and ANN, should we throw an exception?
- if `|KNN| < |ANN|`, this is indicative of either bad KNN/ground-truth data (i.e. we need to fix our test) or the DB erroneously providing too many results
- if `|ANN| < |KNN|`, this is indicative of either bad KNN/ground-truth data (i.e. we need to fix our test) or the DB erroneously providing too few results

Reply:
Updated in latest commit to fix this